photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform PSF Photometry"""
+
+from __future__ import division
+import abc, six
+import numpy as np
+from astropy.table import Table, vstack
+from photutils.psf import subtract_psf
+
+
+__all__ = ['PSFPhotometryBase', 'NStarPSFPhotometry']
+
+
+class PSFPhotometryBase(object):
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def do_photometry(self):
+        pass
+
+
+class NStarPSFPhotometry(PSFPhotometryBase):
+    """
+    This is an implementation of the NSTAR algorithm proposed by Stetson
+    (1987) to perform point spread function photometry in crowded fields.
+
+    This class basically implements the loop FIND, GROUP, NSTAR,
+    SUBTRACT, FIND until no more stars are detected.
+    """
+
+    def __init__(self, find, group, bkg, psf, fitter, niters, fitshape):
+        """
+        Attributes
+        ----------
+        find : an instance of any StarFinderBase subclasses
+        group : an instance of any GroupStarsBase subclasses
+        bkg : an instance of any BackgroundBase2D (?) subclasses
+        psf : Fittable2DModel instance
+        fitter : Fitter instance
+        niters : int
+            number of iterations for the loop FIND, GROUP, SUBTRACT, NSTAR
+        fitshape : array-like
+            rectangular shape around the center of a star which will be used
+            to collect the data to do the fitting
```

Comment:
Instead of `"shape"` keywords, lately I've been using `"size"` keywords (e.g. following `scipy.ndimage` filters) where a scalar size is also allowed (in that case the size is broadcast to use a square region).